src/App.js

```diff
@@ -1,16 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [entriesData, setEntriesData] = useState(chatMessages);
+
+  const calcLikes = (entriesData) => {
+    let likeCount = 0;
+
+    for (const entry of entriesData) {
+      if (entry.liked) {
+        likeCount++;
+      }
+    }
+    return likeCount;
+  };
+
+  const getParticipants = (entriesData) => {
+    let chatMembers = new Set();
+
+    for (const entry of entriesData) {
+      chatMembers.add(entry.sender);
+    }
+
+    return Array.from(chatMembers).join(' and ');
+  };
+
+  const updateEntriesData = (updatedEntry) => {
+    const entriesList = entriesData.map((entry) => {
+      if (entry.id === updatedEntry.id) {
+        return updatedEntry;
+      } else {
+        return entry;
+      }
+    });
+    setEntriesData(entriesList);
+  };
```

Comment:
updateEntriesData executes when the like button on a ChatEntry component is clicked. We really only need to update the heart on a ChatEntry, not the entire message. How would you refactor updateEntriesData so that you aren't updating the entire ChatEntry object?

You can check out this [implementation](https://github.com/ada-c17/react-chatlog/blob/c17/solution/src/App.js#L22-L30) for an idea of how this refactor would look